src/Data/Array/ST.js

```diff
@@ -105,3 +105,11 @@ export const toAssocArrayImpl = function (xs) {
   for (var i = 0; i < n; i++) as[i] = { value: xs[i], index: i };
   return as;
 };
+
+export const push = function (a) {
+   return function (xs) {
+     return function () {
+       return xs.push(a);
+     };
+   };
+ };
```

Comment:
Would you mind using the same `STFn` FFI format used elsewhere here? For example, this could be:

```suggestion
export const pushImpl = function (a, xs) {
  return xs.push(a);
};
```

which would be foreign-imported using `STFn2`, and then `push` would be implemented as:

```
push a = runSTFn2 pushImpl
```

Reply:
Oh sorry, I thought this wasn't done for ST yet.